src/qibosoq/components.py

```diff
@@ -0,0 +1,107 @@
+"""Various helper objects"""
+
+from dataclasses import dataclass
+from enum import IntEnum, auto
+from typing import List, Union
+
+
+@dataclass
+class Config:
+    """General RFSoC Configuration"""
+
+    repetition_duration: int = 100
+    """Time to wait between shots (us)"""
+    adc_trig_offset: int = 200
+    """Time to wait between readout pulse and acquisition (ADC clock ticks)"""
+    reps: int = 1000
+    """Number of shots"""
+
+
+class OperationCode(IntEnum):
+    """Available operations"""
+
+    EXECUTE_PULSE_SEQUENCE = auto()
+    EXECUTE_SWEEPS = auto()
+
+
+@dataclass
+class Qubit:
+    """Qubit object, storing flux information"""
+
+    bias: float = 0.0
+    """Amplitude factor, for sweetspot"""
+    dac: int = None
+    """DAC responsible for flux control"""
+
+
+@dataclass
+class Pulse:
+    """Abstract Pulse object"""
+
+    frequency: float
+    """Freuency of the pulse (MHz)"""
+    amplitude: float
+    """Amplitude factor, multiplied by maximum gain of the DAC"""
+    relative_phase: int
+    """Relative phase (degrees)"""
+    start: float
+    """Start time (us)"""
+    duration: float
+    """Duration of the pulse (us)"""
+
+    name: str
+    """Name of the pulse, typically a serial"""
+    type: str
+    """Can be 'readout', 'drive', 'flux'"""
+
+    dac: int
+    """DAC responsible for firing the pulse"""
+    adc: int
+    """ADC to acquire pulse back, for readout pulses"""
+
+    shape: str
+    """Can be 'rectangular', 'gaussian', 'drag'"""
+    rel_sigma: float = None
+    """Sigma for gaussians and drags, fraction of duration"""
+    beta: float = None
+    """Beta for drag pulses"""
+
+    def __eq__(self, pulse):
+        """Check if two pulses are equal, ignoring start time"""
+        if pulse is None:
+            return False
+        return (
+            self.frequency == pulse.frequency
+            and self.amplitude == pulse.amplitude
+            and self.relative_phase == pulse.relative_phase
+            and self.duration == pulse.duration
+            and self.type == pulse.type
+            and self.dac == pulse.dac
+            and self.shape == pulse.shape
```

Comment:
Instead of enumerating the fields you want to compare, you can rely on the automatic implementation, just telling it which fields you want to skip:
```py
beta: float = field(default=None, compare=False)
```
https://docs.python.org/3/library/dataclasses.html#dataclasses.field

Reply:
Thanks for the suggestion!